### PR TITLE
Throw opm error

### PR DESCRIPTION
**Pull request by joakim-hove**
Wrap construction of all the "big four" in `try catch` blocks.

Messages should be refined - but this should be enough to guarantee that all errors have been logged before leaving opm-common.

OK guys: This was new syntax to ya?!

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp [hunk 1]
```diff
@@ -143,6 +147,16 @@ namespace {
         if (DeckSection::hasSCHEDULE(deck))
             iterateScheduleSection( python, deck.getInputPath(), parseContext, errors, SCHEDULESection( deck ), grid, fp);
     }
+    catch (const OpmInputError& opm_error) {
+        throw;
+    }
+    catch (const std::exception& std_error) {
+        OpmLog::error(fmt::format("An error occured while loading the dynamic properties\n",
```
Comment (atgeirr) [inline src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp:154]: The word loading is a little misleading here I think, hinting at I/O or perhaps parsing trouble. What about "while creating the dynamic simulator schedule"?

File: src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp [hunk 2]
```diff
@@ -1126,40 +1127,46 @@ SummaryConfig::SummaryConfig( const Deck& deck,
...
+        for (const auto& kw : this->m_keywords) {
+            this->short_keywords.insert(kw.keyword());
+            this->summary_keywords.insert(kw.uniqueNodeKey());
+        }
+    }
+    catch (const OpmInputError& opm_error) {
+        throw;
+    }
+    catch (const std::exception& std_error) {
+        OpmLog::error(fmt::format("An error occured while loading the summary properties\n",
```
Comment (atgeirr) [inline src/opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp:1163]: Same here, "loading" is a little confusing.

**Comment by atgeirr** [comment]
> OK guys: This was new syntax to ya?!

I have seen it on functions, but putting the try around the initializer list was new to me. I assume it works as expected. I assume base class initialization is outside the try unless explicit placed inside? (Not relevant here though.)

I have requested a few rewordings of messages. Other than that, I think the changes themselves are in line with what we have discussed, so go ahead. I have one question though: Do we intend to have/keep a try/catch on the main() function or equivalent, and to print the what() string there, or not?

**Comment by joakim-hove** [comment]
> I have requested a few rewordings of messages.

I have made some changes - not easy to really find the sweet spot for these things - I'll let it linger a bit.

> Do we intend to have/keep a try/catch on the main() function or equivalent, and to print the what() string there, or not?

My plan is to keep a `try catch`in the main function - but that is really only to ensure graceful exit in the MPI situation. The message is printed here. See: https://github.com/OPM/opm-simulators/pull/2813 for my take on that end of things.

File: src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp [hunk 3]
```diff
@@ -143,6 +147,16 @@ namespace {
         if (DeckSection::hasSCHEDULE(deck))
             iterateScheduleSection( python, deck.getInputPath(), parseContext, errors, SCHEDULESection( deck ), grid, fp);
     }
+    catch (const OpmInputError& opm_error) {
+        throw;
+    }
+    catch (const std::exception& std_error) {
+        OpmLog::error(fmt::format("An error occured while creating the reservior schedule\n",
```
Comment (atgeirr) [inline src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp:154]: Typo: reservior -> reservoir.

**Comment by atgeirr** [comment]
> I have made some changes - not easy to really find the sweet spot for these things - I'll let it linger a bit.

I think it looks good now (aside from a minor typo).



> My plan is to keep a `try catch`in the main function - but that is really only to ensure graceful exit in the MPI situation. The message is printed here in opm-common where the exception is thrown. See: [OPM/opm-simulators#2813](https://github.com/OPM/opm-simulators/pull/2813) for my take on that end of things.

As long as we keep using exceptions for terminating by letting them propagate (which I think is good), we should have a main try/catch to terminate gracefully, I agree. That catch should not re-print a message that already has been printed. Should we split the main()-ish catch in two, i.e. catching input errors separately, to guarantee this?

**Comment by joakim-hove** [comment]
> Should we split the main()-ish catch in two, i.e. catching input errors separately, to guarantee this?

The input errors are caught in `readDeck()` - so that split is already in place.

**Pull request merged**